components/script/dom/node.rs

```diff
@@ -674,8 +669,15 @@ impl Node {
         self.children_count.get()
     }
 
-    pub(crate) fn ranges(&self) -> &WeakRangeVec {
-        &self.ranges
+    pub(crate) fn ranges(&self) -> Rc<WeakRangeVec> {
+        self.ensure_rare_data().ranges.clone()
```

Comment:
You can use [`Ref::map`](https://doc.rust-lang.org/std/cell/struct.Ref.html#method.map) to return a reference here and avoid having to use an `Rc` for the `WeakRangeVec` (I think)